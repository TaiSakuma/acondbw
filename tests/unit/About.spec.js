import Vue from "vue";
import VueRouter from "vue-router";
import Vuetify from "vuetify";
import { shallowMount, createLocalVue } from "@vue/test-utils";

import About from "@/views/About.vue";

// The way to setup tests with Vuetify was described here
// https://github.com/vuetifyjs/vuetify/issues/4068#issuecomment-586829171

Vue.use(Vuetify);
Vue.use(VueRouter);

const localVue = createLocalVue();

describe("About.vue", () => {
  let vuetify;
  let wrapper;
  beforeEach(() => {
    vuetify = new Vuetify();
    wrapper = shallowMount(About, {
      localVue,
      vuetify
    });
  });

  it("test text", () => {
    expect(wrapper.text()).toBe("This is an about page");
  });

  it("test html", () => {
    expect(wrapper.html()).toContain("<h1>This is an about page</h1>");
  });

  it("match snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
