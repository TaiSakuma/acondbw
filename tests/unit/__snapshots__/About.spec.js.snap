// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About.vue match snapshot 1`] = `
<v-container-stub tag="div" fluid="true" fill-height="">
  <v-row-stub tag="div" align="center" justify="center">
    <h1>This is an about page</h1>
  </v-row-stub>
</v-container-stub>
`;
